File: leetCode/1.ts
function twoSum(nums: number[], target: number, ): number[] {
  const complementMap: Record<number, number> = {}
  for(let idx = 0; idx < nums.length; idx++) {
    const complement = target - nums[idx]
    if(complementMap.hasOwnProperty(complement)) return [ complementMap[complement], idx ]
    complementMap[nums[idx]] = idx
  }

  return []
};

File: leetCode/python/1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
      cMap = {}

      for idx, num in enumerate(nums):
        if (target - num) in cMap:
          return [ idx, cMap[target - num]]
        cMap[num] = idx
  
      return []